bugs/gh-483.txt

```diff
@@ -0,0 +1,425 @@
+   name=$1
+
+   lxc exec $name -- DI_LOG=stderr /usr/lib/cloud-init/ds-identify --force
+   lxc exec $name -- cloud-init clean --logs
+   lxc exec $name -- cloud-init init --local
+   lxc exec $name -- cloud-init init > /dev/null
+   lxc exec $name -- cloud-init modules --mode=config
+   lxc exec $name -- cloud-init modules --mode=final
+   lxc exec $name -- cloud-init status --wait --long
+}
+
+for SERIES in bionic focal xenial; do
+   echo '=== BEGIN ' $SERIES
+   ref=$SERIES-proposed
+   name=test-$SERIES
+   lxc delete $name --force 2> /dev/null
+   lxc launch ubuntu-daily:$SERIES $name -c user.user-data="$(cat ca-certs.yaml)"
+
+   echo "###########################################"
+   echo "Expect cloud-init to finish"
+   lxc exec $name -- cloud-init status --wait --long
+   echo -e "###########################################\n"
+
+   echo "###########################################"
+   echo "Expect ca-certificates.conf first line to be empty"
+   first_line=$(lxc exec $name -- cat /etc/ca-certificates.conf | head -n 1)
+
+   if [ -z "$first_line" ]; then
+       echo "SUCESS: first line of ca-certificates is empty"
+   else
+       echo "FAILURE: first line of ca-certificates is not empty"
+   fi
+   echo -e "###########################################\n"
+
+   echo "###########################################"
+   echo "Reset ca-certificates"
+   lxc exec $name -- rm /etc/ca-certificates.conf
+   lxc exec $name -- rm -rf /usr/share/ca-certificates
+   lxc exec $name -- apt install --reinstall ca-certificates
+   lxc exec $name -- update-ca-certificates
+   lxc exec $name -- sed -i 's/\!//' /etc/ca-certificates.conf
+   lxc exec $name -- dpkg-reconfigure ca-certificates --default-priority
+   echo -e "###########################################\n"
+
+   # upgrade cloud-init to proposed
+   echo "###########################################"
+   echo "Update cloud-init"
+   lxc file push setup_dev_proposed.sh $name/var/tmp/
+   lxc exec $name -- bash /var/tmp/setup_dev_proposed.sh
+   echo -e "###########################################\n"
+
+   echo "###########################################"
+   echo "Restart cloud-init"
+   restart_cloudinit $name
+   echo -e "###########################################\n"
+
+   echo "###########################################"
+   echo "Expect ca-certificates.conf first line to not be empty"
+   first_line=$(lxc exec $name -- cat /etc/ca-certificates.conf | head -n 1)
+
+   if [ "$first_line" = "cloud-init-ca-certs.crt" ]; then
+       echo "SUCESS: first line of ca-certificates is not empty"
+   else
+       echo "FAILURE: first line of ca-certificates is empty"
+   fi
+   echo -e "###########################################\n"
+done
+```
+
+[Regression Potential]
+
+If users rely on that initial blank line in the /etc/ca-certificates.conf file, the code will no
+longer work. However, since this seems to be a behavior that should be avoided, it seems that the
+regression potential is low.
```

Comment:
Worth adding that blank lines should be avoided because they're problematic anyway, as noted in https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/1077020, so it's very unlikely that any deployment relies on them?